src/methods/abstract.js

```diff
@@ -54,11 +56,23 @@ export function SplitMatch(realm: Realm, S: string, q: number, R: string): false
 }
 
 // ECMA262 7.2.1
-export function RequireObjectCoercible<T: Value>(realm: Realm, arg: T): T {
-  if (HasSomeCompatibleType(arg, NullValue, UndefinedValue)) {
+export function RequireObjectCoercible(
```

Comment:
The reason I changed this is that the template does not exclude the possibility of the result being an instance of NullValue or UndefinedValue. The added precision of the other cases is not actually needed anywhere.